logstash-core/lib/logstash/queue_factory.rb

```diff
@@ -24,11 +25,17 @@ def self.create(settings)
         LogStash::Util::WrappedSynchronousQueue.new
       when "persisted"
         # persisted is the disk based acked queue
-        queue_path = settings.get("path.queue")
+        queue_path = ::File.join(settings.get("path.queue"), settings.get("pipeline.id"))
+        create_directory(queue_path)
```

Comment:
I think we could just directly call `mkdir_p` here, since the create directory doesn't do more.
Unless we want the `#create_directory` method do do some error handling, but IIRC permissions to the path will be validated by the settings option when logstash start.